Tests/UpstreamIntegrationTests/UpstreamIntegrationTests.swift

```diff
@@ -0,0 +1,213 @@
+//
+// Copyright 2023 Adobe. All rights reserved.
+// This file is licensed to you under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License. You may obtain a copy
+// of the License at http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software distributed under
+// the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+// OF ANY KIND, either express or implied. See the License for the specific language
+// governing permissions and limitations under the License.
+//
+
+@testable import AEPCore
+import AEPEdge
+import AEPEdgeIdentity
+import AEPServices
+import Foundation
+import XCTest
+
+class NetworkTestingDelegate: NetworkRequestDelegate {
+    var testCaseCompletion: (HttpConnection) -> ()
+    func handleNetworkResponse(httpConnection: AEPServices.HttpConnection) {
+        print("Delegate received httpConnection: \(httpConnection)")
+        testCaseCompletion(httpConnection)
+    }
+    init() {
+        testCaseCompletion = { httpConnection in
+            
+        }
+    }
+}
+
+/// This test class validates proper intergration with upstream services, specifically Edge Network
+class UpstreamIntegrationTests: XCTestCase {
+    private var edgeEnvironment: EdgeEnvironment = .prod
+    private var edgeLocationHint: EdgeLocationHint? = nil
+    
+    private let testingDelegate = NetworkTestingDelegate()
+    
+    let LOG_SOURCE = "SampleFunctionalTests"
+    
+    let asyncTimeout: TimeInterval = 10
+
+    override func setUp() {
+        let networkService = IntegrationTestNetworkService()
+        networkService.testingDelegate = testingDelegate
```

Comment:
When we adopt this pattern to multiple test classes and test cases you can easily forget to set the delegate or to overwrite the testCaseCompletion. I suggest the following workflow:
1. the setup step overwrites the network service with IntegrationTestNetworkService. The IntegrationTestNetworkService can include helpers for registering for httpConnections
2. each test can register a callback for a particular url - e.g. networkService.expect(networkRequest: NetworkRequest completionHandler: ((HttpConnection) -> Void)?)
 - this could be the simplified version where the test case will count and assert on number of invocations internally
 - alternatively we could have something similar with the FunctionalTestNetworkService where you set all the expected network requests and their count and the IntegrationTestNetworkService does the waiting and returns the list of requests / responses (see getReceivedNetworkRequestsMatching, getMatchedResponseForUrlAndHttpMethod)

Let's discuss more about this and take these improvements in a separate task. In the meantime, can you move NetworkTestingDelegate inside IntegrationTestNetworkService to have everything network related in one place?